src/DomainLib.Persistence.EventStore/EventStoreEventsRepository.cs

```diff
@@ -0,0 +1,54 @@
+﻿using EventStore.ClientAPI;
+using EventStore.ClientAPI.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DomainLib.Persistence.EventStore
+{
+    public class EventStoreEventsRepository : IEventsRepository
+    {
+        private readonly IEventStoreConnection _connection;
+        private readonly IEventSerializer _serializer;
+
+        public EventStoreEventsRepository(IEventStoreConnection connection, IEventSerializer serializer)
+        {
+            _connection = connection;
+            _serializer = serializer;
+        }
+
+        public async Task<long> SaveEventsAsync<TEvent>(string streamName, long expectedStreamVersion, IEnumerable<TEvent> events)
+        {
+            var eventDatas = events.Select(e => _serializer.ToEventData(e));
+            
```

Comment:
Should we have an issue to address this TODO?

Reply:
Created https://github.com/daniel-smith/domain-lib/issues/8